plugins/techdocs-node/src/stages/publish/azureBlobStorage.ts

```diff
@@ -78,6 +78,29 @@ export class AzureBlobStoragePublish implements PublisherBase {
       );
     }
 
+    const legacyPathCasing =
+      config.getOptionalBoolean(
+        'techdocs.legacyUseCaseSensitiveTripletPaths',
+      ) || false;
+
+    // Give more priority for connectionString, if configured, return the AzureBlobStoragePublish object here itself
+    const connectionStringKey =
+      'techdocs.publisher.azureBlobStorage.connectionString';
+    const connectionString = config.getOptionalString(connectionStringKey);
+    if (connectionString) {
+      logger.info(
+        `using ${connectionStringKey} string to create BlobServiceClient`,
+      );
+      const storageClient =
+        BlobServiceClient.fromConnectionString(connectionString);
+      return new AzureBlobStoragePublish({
```

Comment:
Just noting that this construction is duplicated now (see below at the bottom of the method). Could have broken out a separate `getBlobServiceClient` or whatever that chooses one of two paths based on whether the connection string was set or not. Or just like

```ts
let blobStorageClient: BlobStorageClient;
if (...connection string is set) {
  blobStorageClient = ...
} else {
  blobStorageClient = ...
}

return new AzureBlobStoragePublish(...);
```